packages/notify-client/src/utils/history.ts

```
@@ -69,18 +71,39 @@ export const fetchAndInjectHistory = async (
   historyClient: HistoryClient
 ) => {
   try {
-    const messages = await historyClient.getMessages({
-      topic,
-      direction: "backward",
-      messageCount: 200,
-    });
-
-    core.logger.info(
-      `Fetched ${messages.messageResponse.messages.length} messages from history`
-    );
+    let originId = "";
+    let retrievedCount = 0;
+
+    // Fetch history until we have exhausted all of this topic's history.
+    do {
+      const baseFetchParams = {
+        topic,
+        direction: "backward",
+        messageCount: HISTORY_MAX_FETCH_SIZE,
+      } as const;
+
+      const fetchParams = originId
+        ? ({
+            ...baseFetchParams,
+            originId,
+          } as const)
+        : baseFetchParams;
+
+      const messages = await historyClient.getMessages(fetchParams);
+
+      core.logger.info(
+        `Fetched ${messages.messageResponse.messages.length} messages from history for topic: ${topic}, store: ${name}`
+      );
+
+      retrievedCount = messages.messageResponse.messages.length;
+      originId =
+        messages.messageResponse.messages[
+          messages.messageResponse.messages.length - 1
+        ].message_id;
 
-    const currentMessages = messages.messageResponse.messages;
-    await reduceAndInjectHistory(core, currentMessages, topic);
+      const currentMessages = messages.messageResponse.messages;
+      await reduceAndInjectHistory(core, currentMessages, topic);
+    } while (retrievedCount === HISTORY_MAX_FETCH_SIZE);
```

Open `do/while` loop is making me a bit nervous for infinite loop possibilities 🤔 

Is there a scenario where `retrievedCount` will be set to `HISTORY_MAX_FETCH_SIZE` but then never update, e.g. if further fetching fails silently?